app/services/hashtag_autocomplete_service.rb

```diff
@@ -297,6 +306,24 @@ def lookup_old(slugs)
 
   private
 
+  def search_using_condition(limited_results, term, type, limit, condition)
+    search_results =
+      search_for_type(type, guardian, term, limit - limited_results.length, condition)
+    return limited_results if search_results.empty?
+
+    return limited_results if !all_data_items_valid?(search_results)
```

Comment:
Why this? If I understand correctly, a single invalid item will cause all of the search results to be ignored. Why don't you just reject the invalid items as you already seem to do on lines 318-320?

Reply:
The 318-320 lines are rejecting any duplicate items for this type. Though I do agree, I could just filter only the valid items, will do that